smk/include/config_parser.smk

```diff
@@ -8,12 +8,246 @@ Authors: Carmen Saenz, Mani Arumugam
+    'TRIMMOMATIC_simple_clip_threshold' : int,
+    'TRIMMOMATIC_threads' : int,
+    'VAMB_GPU' : bool,
+    'VAMB_memory' : int,
+    'VAMB_THREADS' : int,
+    'working_dir' : 'directory'
+}
+
+# Aliases for keys.
+# E.g., This module will always refer to minimum read length with 'READ_minlen'.
+#       But if it cannot find 'READ_minlen', it will look for 'TRIMMOMATIC_minlen'.
+ALIASES = {
+    'READ_minlen' : ['TRIMMOMATIC_minlen'],
+    'FASTP_adapters' : ['FASTP_adaptors'],
+    'TRIMMOMATIC_adaptors' : ['TRIMMOMATIC_adapters']
+}
+
+##############################################
+# Functions to parse the keys
+##############################################
+
+def validate_keytype(config, type_dict, key):
+    # Get all possible aliases
+    pos_keys = [key]
+    if key in ALIASES:
+        pos_keys += ALIASES[key]
+
+    # Check for key in global type registry
+    if key not in type_dict:
+        raise KeyError(f"Configuration key not found in global registry: {key}")
+
+    # Check for all aliases
+    # Get value
+    value = None
+    for x in pos_keys:
+        if x in config:
+            value = config[x]
+            break
+    if value == None:
+        return None
+
+    # Get list of expected types
+    expected_types = type_dict[key]
+    if not isinstance(expected_types, list):
+        expected_types = [expected_types]
+
+    # Set up type discovery
+    valid_type = False
+    found_type = None
+
+    # Check for each possible valid type
+
+    for etype in expected_types:
+
+        # Handle simple cases: etype is inbuilt python type
+        if not isinstance(etype, str):
+            if isinstance(value, etype):
+                valid_type = True
+                found_type = etype
+                break
+
+        # Handle special cases: file and directory
+        if isinstance(etype, str):
+            if etype == 'file':
+                found_type = etype
+                if os.path.isfile(value):
+                    valid_type = True
+            if etype == 'directory':
+                found_type = etype
+                if os.path.isdir(value):
+                    valid_type = True
+
+    # Raise error if right type was not found
+
+    if not valid_type:
+        if found_type == 'directory':
+            raise TypeError(f"Directory mapped to configuration key '{key}' does not exist: {value}")
+        elif found_type == 'file':
+            raise TypeError(f"File mapped to configuration key '{key}' does not exist: {value}")
+        else:
+            raise TypeError("Configuration key '{}' should be of type {}, but got {}".format(key, ' or '.join(x.__name__ for x in expected_types), type(value).__name__))
+
+    # return
+    return(value)
+
+def validate_required_key(config, key):
+
+    # Check if key is present and the right type
+    val = validate_keytype(config, GLOBAL_CONFIG_KEYTYPES, key)
+
+    if val == None:
+        raise KeyError(f"Missing required configuration key: {key}")
+
```

Comment:
The error msg for `None` values of required config parameters should be more clear, and be `f"Missing value for required configuration key: {key}"`

Reply:
Good point. Will fix.